.github/workflows/test.yml

```diff
@@ -49,6 +51,28 @@ jobs:
       - name: lint, formatting
         run: |
           npm run check
+      - name: get emscripten cached location and version
+        run: |
+          echo "emscripten_path=$(pwd)/emsdk" >> $GITHUB_ENV
+          echo "emscripten_version=$(curl https://api.github.com/repos/emscripten-core/emsdk/tags | jq -r '.[0].name')" >> $GITHUB_ENV
```

Comment:
I would just hardcode the version and maybe put it in the build documentation/requirements